Upstream-Status: Inappropriate [configuration specific to the QEMU target]

Configure the U-Boot 2026.01 EFI payload to boot the RAUC script and store
its environment on the first VirtIO disk's FAT partition.

diff --git a/configs/efi-x86_payload64_defconfig b/configs/efi-x86_payload64_defconfig
--- a/configs/efi-x86_payload64_defconfig
+++ b/configs/efi-x86_payload64_defconfig
@@ -12,10 +12,13 @@ CONFIG_FIT=y
 CONFIG_FIT_SIGNATURE=y
 CONFIG_BOOTSTD_FULL=y
 CONFIG_LEGACY_IMAGE_FORMAT=y
+CONFIG_BOOTDELAY=0
 CONFIG_SHOW_BOOT_PROGRESS=y
 CONFIG_USE_BOOTARGS=y
-CONFIG_BOOTARGS="root=/dev/sdb3 init=/sbin/init rootwait ro"
+CONFIG_BOOTARGS=""
+CONFIG_USE_BOOTCOMMAND=y
+CONFIG_BOOTCOMMAND="virtio scan; load virtio 0:1 ${scriptaddr} boot.scr; source ${scriptaddr}"
 CONFIG_SYS_PBSIZE=532
 CONFIG_PRE_CONSOLE_BUFFER=y
 CONFIG_SYS_CONSOLE_INFO_QUIET=y
 CONFIG_DISPLAY_BOARDINFO_LATE=y
@@ -22,7 +25,11 @@ CONFIG_DISPLAY_BOARDINFO_LATE=y
+CONFIG_CMD_BOOTMENU=y
 CONFIG_CMD_IDE=y
 CONFIG_CMD_MMC=y
 CONFIG_CMD_USB=y
-# CONFIG_CMD_SETEXPR is not set
+CONFIG_CMD_SETEXPR=y
+CONFIG_CMD_VIRTIO=y
+CONFIG_CMD_NVEDIT_LOAD=y
+CONFIG_CMD_SAVEENV=y
 CONFIG_BOOTP_BOOTFILESIZE=y
 CONFIG_CMD_TIME=y
 CONFIG_CMD_EXT4_WRITE=y
@@ -29,7 +36,7 @@ CONFIG_CMD_EXT4_WRITE=y
 CONFIG_MAC_PARTITION=y
 CONFIG_ENV_OVERWRITE=y
 CONFIG_ENV_IS_IN_FAT=y
-CONFIG_ENV_FAT_INTERFACE="scsi"
+CONFIG_ENV_FAT_INTERFACE="virtio"
 CONFIG_ENV_FAT_DEVICE_AND_PART="0:1"
 CONFIG_ENV_RELOC_GD_ENV_ADDR=y
 CONFIG_USE_BOOTFILE=y
@@ -49,3 +56,10 @@ CONFIG_SYS_64BIT_LBA=y
 # CONFIG_PCI_PNP is not set
 CONFIG_SYS_NS16550_PORT_MAPPED=y
 # CONFIG_GZIP is not set
+CONFIG_BLK=y
+CONFIG_PCI=y
+CONFIG_VIRTIO=y
+CONFIG_VIRTIO_BLK=y
+CONFIG_VIRTIO_PCI=y
+CONFIG_FS_FAT=y
+CONFIG_FAT_WRITE=y
